src/timely-util/src/containers.rs

```diff
@@ -229,14 +265,32 @@ mod builder {
             let words = self.current.borrow().length_in_words();
             let round = (words + ((1 << 18) - 1)) & !((1 << 18) - 1);
             if round - words < round / 10 {
-                let mut alloc = Vec::with_capacity(round);
-                columnar::bytes::serialization::encode(
-                    &mut alloc,
-                    self.current.borrow().as_bytes(),
-                );
-                self.pending
-                    .push_back(Column::Align(alloc.into_boxed_slice()));
-                self.current.clear();
+                /// Move the contents from `current` to an aligned allocation, and push it to `pending`.
+                /// The contents must fit in `round` words (u64).
+                #[cold]
+                fn outlined_align<C>(
+                    current: &mut C::Container,
+                    round: usize,
+                    pending: &mut VecDeque<Column<C>>,
+                ) where
+                    C: Columnar,
+                {
+                    let mut alloc = alloc_aligned_zeroed(round);
+                    let mut writer = std::io::Cursor::new(bytemuck::cast_slice_mut(&mut alloc[..]));
+                    for (align, bytes) in current.borrow().as_bytes() {
+                        assert!(align <= 8);
+                        let length = u64::cast_from(bytes.len());
+                        let length_slice = bytemuck::cast_slice(std::slice::from_ref(&length));
+                        writer.write_all(length_slice).unwrap();
+                        writer.write_all(bytes).unwrap();
+                        let padding = usize::cast_from((8 - (length % 8)) % 8);
+                        writer.write_all(&[0; 8][..padding]).unwrap();
```

Comment:
Is there a `columnar` method we could invoke so we don't have to worry about this code diverging from the `columnar` implementation?